src/utils/viewer/viewer.cpp

```diff
@@ -213,130 +189,75 @@ Viewer::Viewer(const Arguments& arguments)
-  stageAttributes->setRequiresLighting(info.requiresLighting);
-
+  sceneFileName = args.value("scene");
   bool useBullet = args.isSet("enable-physics");
-  // construct physics manager based on specifications in attributes
-  resourceManager_.initPhysicsManager(physicsManager_, useBullet, rootNode_,
-                                      physicsManagerAttributes);
-
-  std::vector<int> tempIDs{sceneID, esp::ID_UNDEFINED};
-  bool stageLoadSuccess = resourceManager_.loadStage(
-      stageAttributes, physicsManager_, &sceneManager_, tempIDs, false);
-  if (!stageLoadSuccess) {
-    LOG(FATAL) << "cannot load " << stageFileName;
-  }
   if (useBullet && (args.isSet("debug-bullet"))) {
     debugBullet_ = true;
   }
 
-  const Mn::Range3D& sceneBB = rootNode_->computeCumulativeBB();
-  resourceManager_.setLightSetup(esp::gfx::getLightsAtBoxCorners(sceneBB));
+  // configure and intialize Simulator
+  auto simConfig = esp::sim::SimulatorConfiguration();
+  simConfig.scene.id = sceneFileName;
+  simConfig.enablePhysics = useBullet;
+  simConfig.frustumCulling = true;
+
+  simulator_ = esp::sim::Simulator::create_unique(simConfig);
+
+  objectAttrManager_ = simulator_->getObjectAttributesManager();
+  assetAttrManager_ = simulator_->getAssetAttributesManager();
+  stageAttrManager_ = simulator_->getStageAttributesManager();
+  physAttrManager_ = simulator_->getPhysicsAttributesManager();
+
+  // configure and initialize default Agent and Sensor
+  auto agentConfig = esp::agent::AgentConfiguration();
+  agentConfig.height = rgbSensorHeight;
+  agentConfig.actionSpace = {
+      // setup viewer action space
+      {"moveForward",
+       esp::agent::ActionSpec::create(
+           "moveForward",
+           esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"moveBackward",
+       esp::agent::ActionSpec::create(
+           "moveBackward",
+           esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"moveLeft",
+       esp::agent::ActionSpec::create(
+           "moveLeft", esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"moveRight",
+       esp::agent::ActionSpec::create(
+           "moveRight", esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"moveDown",
+       esp::agent::ActionSpec::create(
+           "moveDown", esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"moveUp",
+       esp::agent::ActionSpec::create(
+           "moveUp", esp::agent::ActuationMap{{"amount", moveSensitivity}})},
+      {"turnLeft",
+       esp::agent::ActionSpec::create(
+           "turnLeft", esp::agent::ActuationMap{{"amount", lookSensitivity}})},
+      {"turnRight",
+       esp::agent::ActionSpec::create(
+           "turnRight", esp::agent::ActuationMap{{"amount", lookSensitivity}})},
+      {"lookUp",
+       esp::agent::ActionSpec::create(
+           "lookUp", esp::agent::ActuationMap{{"amount", lookSensitivity}})},
+      {"lookDown",
```

Comment:
We should probably also add the default actions added in #676

Reply:
Could bind this to keys, but haven't done so for viewer before.

Reply:
These are not defined in C++. If we add them we can expose in Viewer then.

Reply:
Ah, my bad. Don't bother then.